src/coreComponents/physicsSolvers/fluidFlow/ThermalSinglePhaseFVMKernels.hpp

```diff
@@ -476,6 +476,353 @@ class FaceBasedAssemblyKernelFactory
   }
 };
 
+/******************************** DirichletFaceBasedAssemblyKernel ********************************/
+
+/**
+ * @class DirichFaceBasedAssemblyKernel
+ * @tparam FLUIDWRAPPER the type of the fluid wrapper
+ * @brief Define the interface for the assembly kernel in charge of Dirichlet face flux terms
+ */
+template< integer NUM_DOF, typename FLUIDWRAPPER >
+class DirichletFaceBasedAssemblyKernel : public singlePhaseFVMKernels::DirichletFaceBasedAssemblyKernel< NUM_DOF, FLUIDWRAPPER >
+{
+public:
+
+/**
+ * @brief The type for element-based data. Consists entirely of ArrayView's.
+ *
+ * Can be converted from ElementRegionManager::ElementViewConstAccessor
+ * by calling .toView() or .toViewConst() on an accessor instance
+ */
+  template< typename VIEWTYPE >
+  using ElementViewConst = ElementRegionManager::ElementViewConst< VIEWTYPE >;
+
+  using AbstractBase = singlePhaseFVMKernels::FaceBasedAssemblyKernelBase;
+  using DofNumberAccessor = AbstractBase::DofNumberAccessor;
+  using PermeabilityAccessors = AbstractBase::PermeabilityAccessors;
+  using SinglePhaseFlowAccessors = AbstractBase::SinglePhaseFlowAccessors;
+  using SinglePhaseFluidAccessors = AbstractBase::SinglePhaseFluidAccessors;
+
+  using AbstractBase::m_dt;
+  using AbstractBase::m_rankOffset;
+  using AbstractBase::m_dofNumber;
+  using AbstractBase::m_ghostRank;
+  using AbstractBase::m_gravCoef;
+  using AbstractBase::m_mob;
+  using AbstractBase::m_pres;
+  using AbstractBase::m_permeability;
+  using AbstractBase::m_dPerm_dPres;
+
+  using AbstractBase::m_localMatrix;
+  using AbstractBase::m_localRhs;
+
+  using Base = singlePhaseFVMKernels::DirichletFaceBasedAssemblyKernel< NUM_DOF, FLUIDWRAPPER >;
+  using Base::numDof;
+  using Base::numEqn;
+  using Base::m_stencilWrapper;
+  using Base::m_seri;
+  using Base::m_sesri;
+  using Base::m_sei;
+  using Base::m_facePres;
+  using Base::m_faceGravCoef;
+
+  using ThermalSinglePhaseFlowAccessors =
+    StencilAccessors< fields::flow::temperature,
+                      fields::flow::dMobility_dPressure,
```

Comment:
I don't think you need `dMobility_dPressure` here, it should already be in `SinglePhaseFlowAccessors`
```suggestion
```

Reply:
removed